ivadomed/evaluation.py

```diff
@@ -88,6 +93,16 @@ def evaluate(bids_df, path_output, target_suffix, eval_params):
         nib_painted = nib.Nifti1Image(data_painted, nib_pred.affine)
         nib.save(nib_painted, fname_paint)
 
+        # Write painted PNG if subject file is not NifTI
+        # For Microscopy PNG/TIF files
+        # TODO: implement OMETIFF behavior (elif "ome" in extension)
+        if "nii" not in extension:
+            painted_list = imed_inference.split_classes(nib_painted)
+            imed_inference.pred_to_png(painted_list,
+                                       target_suffix,
+                                       os.path.join(path_preds, subj_acq),
+                                       painted=True)
+
```

Comment:
I also noticed that for the test suite, only `test_testing.py` was modified, so that also leads me to believe that `testing.py` is the right place for the png-generating code.

Reply:
The 2 calls for the function write different files. The first one in `test` writes the prediction from the network to PNG (`_pred`). In evaluate, the prediction is compared to the ground-truth and "painted" according to true positives, false positives and false negatives. So the second call is to write PNG of these painted images (`_pred_painted`).

As for the test, I only wanted to test the function once, and it was better suited with the already existing 2D test in `test_testing.py`.

Reply:
> The 2 calls for the function write different files. 

Oh! My apologies, this is exactly what I was missing. :facepalm: